core/src/main/java/org/apache/carbondata/core/index/TableIndex.java

```diff
@@ -207,15 +208,22 @@ public CarbonTable getTable() {
       Map<Segment, List<Index>> indexes) throws IOException {
     Set<String> missingSISegments = filter.getMissingSISegments();
     for (Segment segment : segments) {
+      List<Index> segmentIndices = indexes.get(segment);
       if (segment == null ||
-          indexes.get(segment) == null || indexes.get(segment).isEmpty()) {
+          indexes.get(segment) == null || segmentIndices.isEmpty()) {
```

Comment:
In line 213, `indexes.get(segment)` is called again. Avoid it by using the existing result from `segmentIndices`.

Reply:
done